File: python/165_比较版本号_medium.py
class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:
        for x, y in zip_longest(version1.split('.'), version2.split('.'), fillvalue=0):
            x = int(x)
            y = int(y)
            if x != y:
                return 1 if x > y else -1
        return 0

File: cpp/165_比较版本号_medium.cpp
class Solution {
public:
    int compareVersion(string version1, string version2) {
        int n = version1.length(), m = version2.length();
        int i = 0, j = 0;
        while (i < n || j < m) {
            int x = 0;
            for (; i<n && version1[i] != '.'; i++) {
                x = x * 10 + version1[i] - '0';
            }
            i++;
            int y = 0;
            for (; j<m && version2[j] != '.'; j++) {
                y = y * 10 + version2[j] - '0';
            }
            j++;
            if (x != y) {
                return x > y ? 1 : -1;
            }
        }
        return 0;
    }
};